torch/testing/_internal/common_device_type.py

```diff
@@ -1230,6 +1237,12 @@ def __init__(self, dep, reason):
         super().__init__(dep, reason, device_type="xpu")
 
 
+# Skips a test on XPU or CUDA if the condition is true.
+class skipGPUIf(skipIf):
+    def __init__(self, dep, reason):
+        super().__init__(dep, reason, device_type=("xpu", "cuda"))
```

Comment:
Could you try to generalize device_type=("xpu", "cuda") to is_gpu(device_type) so that the hard code is relatively less spread out.
is_gpu is in torch.inductor.utils